client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -151,7 +150,7 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
-		if ( this.props.transaction.step.name === transactionStepTypes.SUBMITTING_WPCOM_REQUEST ) {
+		if ( includes( [ transactionStepTypes.SUBMITTING_WPCOM_REQUEST, transactionStepTypes.RECEIVED_WPCOM_RESPONSE ], this.props.transaction.step.name ) ) {
```

Comment:
Well, the function name `redirectIfEmptyCart` kind of explains it by itself, but we can if you want.

```
// Do not redirect if cart is empty, but we're submitting or received transaction response
```